src/main/java/org/edumips64/utils/cli/StepCommand.java

```diff
@@ -0,0 +1,41 @@
+package org.edumips64.utils.cli;
+
+import org.edumips64.utils.CurrentLocale;
+import picocli.CommandLine.Command;
+import picocli.CommandLine.ParentCommand;
+import picocli.CommandLine.Parameters;
+import picocli.CommandLine.Option;
+
+@Command(name = "step", resourceBundle = "CliMessages")
+public class StepCommand implements Runnable {
+
+    @ParentCommand
+    private Cli cli;
+
+    @Option(names ={"-h", "--help"}, usageHelp = true, descriptionKey = ".help.usage.description")
+    private boolean help;
+
+    @Parameters(defaultValue = "1", descriptionKey = "steps.parameter")
+    private Integer steps;
+
+    @Override
+    public void run() {
+        try {
+            if (steps > 0) {
+                printStepMsg();
+            }
+
+            for (int i = 0; i < steps; ++i) {
+                cli.getCPU().step();
```

Comment:
This will throw an exception if the CPU is not in the right step (try loading the JAR and trying to `step` without opening a program).

You should check for the CPU status, similarly to this: https://github.com/EduMIPS64/edumips64/blob/master/src/main/java/org/edumips64/client/Simulator.java#L85

I recognize that this is sub-optimal and the interface is not great... could be easier! The CPU could provide a method to let the caller know that it can run code. But to be fair the CPU class was never designed with multiple possible simulator implementations!

Reply:
Ahh.. I see. I will modify it to check for the status!